desc/magnetic_fields.py

```diff
@@ -795,3 +831,464 @@ def odefun(rpz, s):
+        theta = surface_grid.nodes[:, 1]
+        zeta = surface_grid.nodes[:, 2]
+        ns_mag = data["|e_theta x e_zeta|"]
+        # compute the dV element for the surface needed for biot savart
+        self._dV = (
+            surface_grid.weights * ns_mag / surface_grid.NFP
+        )  # divide by NFP here so that the NFP loop
+        # in compute_magnetic_field is correct
+
+        # compute the potential derivatives on the surface
+        phi_t = self._potential_t(theta, zeta)
+        phi_z = self._potential_z(theta, zeta)
+
+        # compute K = n x nabla(Phi)
+        self._K = xyz2rpz_vec(
+            -(phi_t * (1 / ns_mag) * data["e_zeta"].T).T
+            + (phi_z * (1 / ns_mag) * data["e_theta"].T).T,
+            phi=self.surface_grid.nodes[:, 2],
+        )
+
+    def compute_magnetic_field(
+        self, coords, params=None, basis="rpz", surface_grid=None
+    ):
+        """Compute magnetic field at a set of points.
+
+        Parameters
+        ----------
+        coords : array-like shape(N,3) or Grid
+            cylindrical or cartesian coordinates
+        params : dict, optional
+            parameters to pass to current potential function
+        basis : {"rpz", "xyz"}
+            basis for input coordinates and returned magnetic field
+        surface_grid : Grid,
+            grid upon which to evaluate the surface current density K
+
+        Returns
+        -------
+        field : ndarray, shape(N,3)
+            magnetic field at specified points
+
+        """
+        assert basis.lower() in ["rpz", "xyz"]
+        if isinstance(coords, Grid):
+            coords = coords.nodes
+        coords = jnp.atleast_2d(coords)
+        if basis == "rpz":
+            coords = rpz2xyz(coords)
+
+        if (params is None) or (len(params) == 0):
+            params = self._params
+        # compute surface current, and store grid quantities
+        # needed for integration in class
+        self._compute_surface_current(surface_grid=surface_grid, params=params)
+
+        def nfp_loop(j, f):
+            # calculate (by rotating) rs, rs_t, rz_t
+            phi = (
+                self.surface_grid.nodes[:, 2] + j * 2 * jnp.pi / self.surface_grid.NFP
+            ) % (2 * jnp.pi)
+            rs = rpz2xyz_vec(self._rs, phi=phi)
+            K = rpz2xyz_vec(self._K, phi=phi)
+            fj = biot_savart_general(
+                coords,
+                rs,
+                K,
+                self._dV,
+            )
+            f += fj
+            return f
+
+        B = fori_loop(0, self.surface_grid.NFP, nfp_loop, jnp.zeros_like(coords))
+        if basis == "rpz":
+            B = xyz2rpz_vec(B, x=coords[:, 0], y=coords[:, 1])
+        return B
+
+
+class FourierCurrentPotentialField(CurrentPotentialField):
+    """Magnetic field due to a surface current potential on a toroidal surface.
+
+        surface current K is assumed given by
+
+        K = ∇ Φ x n
```

Comment:
should check whether its this or the flipped, with minus sign

Reply:
It is this, this is opposite the REGCOIL convention, I chose this though so that positive G means positive toroidal field, which is more intuitive to me

![image](https://github.com/PlasmaControl/DESC/assets/37969854/92a2708f-4998-4ec9-be4b-82215da7ca83)

Reply:
I'd prefer `n x grad(Phi)` since that seems to be much more common in the literature and other codes. It's also what I've been using on the free boundary branch.

Reply:
Sure, I can change it, but begrudgingly as I don't like that a positive poloidal current `G` results in a negative `B_zeta` now